userspace/libsinsp/test/prefix_search.ut.cpp

```diff
@@ -0,0 +1,309 @@
+/*
+Copyright (C) 2023 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include <gtest/gtest.h>
+
+#include <prefix_search.h>
+
+using namespace std;
+
+TEST(prefix_search_test, basic)
+{
+	path_prefix_search tree;
+
+	tree.add_search_path("/var/run");
+	tree.add_search_path("/var/run/dmesg");
+	tree.add_search_path("/etc/");
+	tree.add_search_path("/lib");
+	tree.add_search_path("/usr/lib");
+	tree.add_search_path("/usr/local");
+
+	bool found = tree.match("/var/run/docker");
+	ASSERT_TRUE(found);
+
+	found = tree.match("/boot");
+	ASSERT_FALSE(found);
+
+	found = tree.match("/var/lib/messages");
+	ASSERT_FALSE(found);
+
+	found = tree.match("/var");
+	ASSERT_FALSE(found);
+
+	found = tree.match("/var/run");
+	ASSERT_TRUE(found);
+
+	found = tree.match("/usr");
+	ASSERT_FALSE(found);
+}
+
+TEST(prefix_search_test, glob)
+{
+	path_prefix_search tree;
+
+	tree.add_search_path("/opt/*/subdir");
+	tree.add_search_path("/opt2/*");
+	tree.add_search_path("/opt3/*/subdir2/*");
+
+	bool found = tree.match("/opt");
+	ASSERT_FALSE(found);
+
+	found = tree.match("/opt/dir/subdir");
+	ASSERT_TRUE(found);
+
+	found = tree.match("/opt/dir/subdir/file.txt");
+	ASSERT_TRUE(found);
+
+	found = tree.match("/opt3/dir/subdir2");
+	ASSERT_FALSE(found);
+
+	found = tree.match("/opt3/dir/subdir2/file.txt");
+	ASSERT_TRUE(found);
+}
+
+TEST(prefix_search_test, subpaths)
```

Comment:
I like these test cases. Can we, even just for a simple example, have a comparison with a full example of `tree.as_string()` output? That would help us catch representation regressions in the future.

Reply:
Sure, will add.